github/github.go

```diff
@@ -224,51 +225,78 @@ func (c *Client) ListRunners(ctx context.Context, enterprise, org, repo string)
 	return runners, nil
 }
 
-func (c *Client) GetRunnerGroupsFromRepository(ctx context.Context, org, repo string, potentialEnterpriseGroups []string, potentialOrgGroups []string) ([]string, []string, error) {
+func (c *Client) GetRunnerGroupsFromRepository(ctx context.Context, org, repo string, potentialGroups utils.RunnerGroups) (utils.RunnerGroups, error) {
 
-	var enterpriseRunnerGroups []string
-	var orgRunnerGroups []string
+	var visibleGroups utils.RunnerGroups
 
 	if org != "" {
 		runnerGroups, err := c.getOrganizationRunnerGroups(ctx, org, repo)
```

Comment:
`repo` seems redundant (apparently unused within the function).

Also, (not your fault at all but)the name of this function(which is seemingly derived from the `ListOrganizationRunnerGroups` function provided by go-github) doesn't capture the fact that it also returns enterprise runners that are inherited to the organization.

That said, changing this to something like the below would improve the readability of code:

```suggestion
		runnerGroups, err := c.getDefinedAndInheritedRunnerGroups(ctx, org)
```